File: Android/2016-04-29-ButterKnifeExample/app/src/main/java/tech/thdev/butter_knife_example/network/PhotoServiceInterface.java
package tech.thdev.butter_knife_example.network;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import tech.thdev.butter_knife_example.BuildConfig;
import tech.thdev.butter_knife_example.network.domain.RecentPhotoResponse;

/**
 * Created by Tae-hwan on 5/3/16.
 */
public interface PhotoServiceInterface {

    @GET("?format=json&nojsoncallback=1&method=flickr.interestingness.getList&api_key=" + BuildConfig.FLICKER_API_KEY)
    Call<RecentPhotoResponse> getFlickrPhotos(
            @Query("page") int page);

    @GET("?method=flickr.photos.search&safe_search=1&format=json&nojsoncallback=1&api_key=" + BuildConfig.FLICKER_API_KEY)
    Call<RecentPhotoResponse> getSearchPhotoResponse(
            @Query("per_page") int perPage,
            @Query("page") int page,
            @Query("text") String text);

    @GET("?method=flickr.photos.getRecent&format=json&nojsoncallback=1&api_key=" + BuildConfig.FLICKER_API_KEY)
    Call<RecentPhotoResponse> getRecentPhotoResponse(
            @Query("per_page") int perPage,
            @Query("page") int page);
}

File: kotlin-example-01/src/main/kotlin/tech/thdev/kotlin_example_01/network/PhotoServiceInterface.kt
package tech.thdev.kotlin_example_01.network

import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable
import tech.thdev.kotlin_example_01.BuildConfig
import tech.thdev.kotlin_example_01.model.PhotoResponse

/**
 * Created by Tae-hwan on 7/22/16.
 */
interface PhotoServiceInterface {

    @GET("?method=flickr.interestingness.getList&format=json&nojsoncallback=1&api_key=" + BuildConfig.FLICKR_API_KEY)
    fun getRecentFlickrPhotos(@Query("page") page: Int): Observable<PhotoResponse>

    @GET("?method=flickr.photos.search&format=json&nojsoncallback=1&apk_key=" + BuildConfig.FLICKR_API_KEY)
    fun getFlickrPhotosSearch(
        @Query("page") page: Int,
        @Query("safe_search") safeSearch: Int,
        @Query("text") text: String): Observable<PhotoResponse>
}
